WordPress/openverse-catalog
# Refactor Europeana to use `ProviderDataIngester`

## Fixes


Fixes #584 by @stacimc 

## Description



1. Refactors Europeana's provider script to use the ProviderDataIngester base class.
2. Adds a simple default implementation for `get_media_type` to the base class to remove boilerplate for providers that only have a single media type. This seemed like an easy improvement to me but there may be a reason why this isn't a good default implementation.

## Testing Instructions


Checkout the unit tests. Many of them had to be removed and new ones added because they were testing functionality that doesn't exist anymore (is abstracted away) or had to test new functionality. The existing tests that pertained to the data transformation I tried to keep as close as possible to the original tests. I caught some bugs in my refactor this way, which was very helpful! I did not add any new data transformation tests as the existing ones seemed comprehensive enough to me.

Finally, run the DAG. Follow the instructions in the README to run Airflow locally and then run the DAG. You can run it without an API key for testing. Currently this doesn't work though, at least not for me locally, due to an issue with the filename generation task. cc @stacimc for help debugging this. Here's the log output from the task:

```
[2022-10-24, 04:38:11 UTC] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 193, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/usr/local/airflow/openverse_catalog/dags/providers/factory_utils.py", line 66, in generate_tsv_filenames
    stores = _load_provider_script_stores(ingestion_callable, media_types)
  File "/usr/local/airflow/openverse_catalog/dags/providers/factory_utils.py", line 37, in _load_provider_script_stores
    raise ValueError(
ValueError: Expected stores in providers.provider_api_scripts.europeana were missing: ['image']
```

I haven't dug into this yet but it seems like probably a configuration issue on my end.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
